suzieq/poller/worker/services/macs.py

```diff
@@ -45,9 +45,12 @@ def _add_mackey_protocol(self, entry):
                     entry['mackey'] = entry['oif']
             else:
                 if entry.get('vlan', 0):
-                    entry['mackey'] = entry['vlan']
+                    if entry['flags'] in ['static', 'permanent', 'router']:
+                        entry['mackey'] = f'{entry["vlan"]}-{entry["oif"]}'
+                    else:
+                        entry['mackey'] = entry['vlan']
                 else:
-                    if entry['flags'] in ['static', 'permanent']:
+                    if entry['flags'] in ['static', 'permanent', 'router']:
```

Comment:
What about older records? Don't we need some kind of "migration function" to update the mackeys?

Reply:
Excellent question. The bug was introduced in 0.18.0 and only affects NXOS. Its not a critical bug in the sense that none of the critical functionality will be broken as a consequence of this. I could provide a migration function, I suppose.